### PR TITLE
Update to v0.11

### PR DESCRIPTION
**Pull request by yminsky**
This updates to the v0.11 version of Base and other JS libraries. Also it adds some non-deterministic markers.

### CODE REVIEW

**Comment by avsm** [comment]
Looks like another change is needed for rwo_toc:

```
File "bin/lib/rwo_toc.ml", line 202, characters 21-63:
Error: The function applied to this argument has type
         cmp:('a -> 'a -> int) -> 'a list -> 'a list
This argument cannot be applied with label ~compare
```

File: examples/code/command-line-parsing/basic_md5_with_optional_file_broken/build_basic_md5_with_optional_file_broken.errsh [hunk 1]
```diff
@@ -1,8 +1,8 @@
...
   File "basic_md5_with_optional_file_broken.ml", line 17, characters 4-11:
-  Error: This expression has type string -> unit -> unit
+  Error: This expression has type Base.string -> unit -> unit
```
Comment (avsm) [inline examples/code/command-line-parsing/basic_md5_with_optional_file_broken/build_basic_md5_with_optional_file_broken.errsh]: It's a pity short paths doesnt convert this Base.string to string

File: examples/code/error-handling/exn_cost/run_exn_cost.sh [hunk 2]
```diff
@@ -1,39 +1,11 @@
...
-  File "exn_cost.ml", line 10, characters 33-37:
-  Warning 3: deprecated: Exit
-  [2016-09] this element comes from the stdlib distributed with OCaml.
-  Refering to the stdlib directly is discouraged by Base. You should either
```
Comment (avsm) [inline examples/code/error-handling/exn_cost/run_exn_cost.sh]: I notice this typo is still present in the latest Base (`refering` -> `referring`)

File: examples/code/command-line-parsing/basic_md5_with_optional_file_broken/build_basic_md5_with_optional_file_broken.errsh (hunk 1 above)
Comment (yminsky) [reply examples/code/command-line-parsing/basic_md5_with_optional_file_broken/build_basic_md5_with_optional_file_broken.errsh]: When Leo's short-paths patch lands, this issue should be resolved.

**Comment by yminsky** [comment]
Maybe you were on the wrong version when building? rwo_toc.ml is already fixed in this feature, and I verified that it builds.

**Comment by samoht** [comment]
CI cache needs to be updated manually whenever the dependencies are changing by doing:

```
make docker
docker push ocaml/rwo
```

**Comment by samoht** [comment]
Note this breaks the compilation of examples, as `async` examples are using `cohttp-async` which do not compile with base v0.11.0 yet.

**Pull request merged**